src/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.tsx

```diff
@@ -87,8 +87,10 @@ export const SignStep = ({
     isLastTransaction && !waitingForDevice ? 'Sign & Submit' : signBtnLabel;
   signBtnLabel = continueWithoutSigning ? 'Continue' : signBtnLabel;
 
+  const nftId = `${tokenId}-${nonce}`;
+
   const { tokenDecimals, tokenAvatar } = useGetTokenDetails({
-    tokenId: currentTransaction.transactionTokenInfo.tokenId
+    tokenId: nonce ? nftId : tokenId
```

Comment:
this will do a false positive if nonce is 0, is this what we want? or do we need to do a nonce != null?

Reply:
Also I think we should add a little commentary here, it's a bit confusing what this does

Reply:
the nonce is a string

Reply:
done